src/features/wallet/hooks.tsx

```diff
@@ -64,29 +81,50 @@ export function useAccounts(): {
   const solAccountInfo: AccountInfo = useMemo(
     () => ({
       protocol: ProtocolType.Sealevel,
-      address: solAddress,
+      addresses: solAddress ? [{ address: solAddress }] : [],
       connectorName: solConnectorName,
       isReady: isSolAccountReady,
     }),
     [solAddress, solConnectorName, isSolAccountReady],
   );
 
   // Cosmos
-  const {
-    address: cosmAddress,
-    wallet: cosmWallet,
-    isWalletConnected: isCosmAccountReady,
-  } = useCosmosChain(getCosmosChainName());
+  const { wallet: cosmWallet, isWalletConnected: isCosmAccountReady } =
+    useCosmosChain(PLACEHOLDER_COSMOS_CHAIN);
+  const { status: cosmWalletStatus, client: cosmWalletClient } = useCosmosWallet();
+  const [cosmAccounts, setCosmAccounts] = useState<Array<ChainAddress>>([]);
+
+  useEffect(() => {
+    if (cosmWalletStatus !== 'Done' || !cosmWalletClient) return;
+    const multiProvider = getMultiProvider();
+    const cosmChainNames = getCosmosChainNames();
+    const cosmChainCaip2Ids = cosmChainNames.map((c) =>
+      getCaip2Id(ProtocolType.Cosmos, multiProvider.getChainId(c)),
+    );
+    cosmWalletClient.enable?.(cosmChainNames).catch((err) => logger.error(err));
+    const cosmosAccountPromises = cosmChainNames.map((c) => cosmWalletClient.getAccount?.(c));
+    Promise.all(cosmosAccountPromises)
```

Comment:
Does this not need to be awaited?

Reply:
You can't await in a useEffect. It has a `then` fn attached to it